src/bind.py

```diff
@@ -184,30 +184,49 @@ def _generate_named_conf_local(self, zones: typing.List[str]) -> str:
             )
         return content
 
-    def handle_new_relation_data(self, rrd: DNSRecordRequirerData) -> DNSRecordProviderData:
+    def handle_relation_data(
+        self,
+        relation_data: typing.List[typing.Tuple[DNSRecordRequirerData, DNSRecordProviderData]],
+    ) -> DNSRecordProviderData:
         """Handle new relation data.
 
         Args:
-            rrd: The DNSRecordRequirerData from the relation
+            relation_data: The list of DNSRecordRequirerData from the dns_record relations
 
         Returns:
             A resulting DNSRecordProviderData to put in the relation databag
         """
-        zones = self._to_bind_zones(rrd)
-        logger.debug("ZONES: %s", zones)
+        # Create staging area
         tempdir = tempfile.mkdtemp()
+
+        # Write zone files
+        zones = {}
+        for rrd, _ in relation_data:
```

Comment:
rrd?

Reply:
that was for record_requirer_data, I replaced it